README.md
# FormDesigner
FormDesigner for project VisualuRuby

## 修正内容
- Exerb でEXE化可能な様に修正
- 出力するコントロールの追加内容をソートする(毎回同じになる)ように修正

## オリジナルファイルについて
- FormDesigner for Project VisualuRuby Ver 060501
- https://ja.osdn.net/projects/fdvr/
- Subversion リポジトリ r71(r65以降)の/formdesigner/trunk を使用
- https://ja.osdn.net/projects/fdvr/svn/view/formdesigner/trunk/?root=fdvr
- 開発者:雪見酒
- ライセンス：Ruby Licence

### Exerb
- http://exerb.osdn.jp/man/README.ja.html
- 開発者:加藤勇也
- ライセンス：LGPL


history.txt
#history of FormDesigner for VisualuRuby

2003/10/13
          FormDesiner 031013 released
2003/10/15
          Updated readme.txt
          Removed vruby030104unstable.zip
2003/10/19
          Corrected the problem when changing the attribute of the
          control on "Rebar."
2003/10/20
          Enabled it to choose a new-line code from two, LF and CRLF,
          on cygwin.
2003/10/26
          Fixed the problem of "UpdateProject" when there is a 
          ListView.
2003/11/02
          Changed interface of menu items editing.
          Useabled popup menu.
2003/11/04
          FormDesiner 031103 released
2003/11/05
          Fixed the stupid bug which will be downed if "Version" is 
          clicked... ah
2003/11/06
          Corrected becoming an error, when the file of an old version
           with a menu was read.
2003/11/10
          Fixed the bug which will fall down if a button is pushed
          without choosing anything in the menu editor.
2003/11/14
          Corrected that up-and-down operation did not work in the 
          sub menu of a menu editor.
2003/11/16
          Added the copyrighs to the source code.
2003/11/17
          Fixed some bugs:
          If the attribute of the control on Rebar is changed, it will down 
          at the time of a source output.
          It will down, if a list view is put on TabbedPanel and a style 
          is changed.
          If ListView is on TbbedPanel, other control will be missing 
          at the time of a source output.
2004/02/25
          Corrected the bug which becomes an error, when the caption of 
          control the quotation mark was contained.
          Moved the owner draw to the attribute tab from the style dialog.
          Cut,copy and paste were mounted.
          EditComboBox was added.
          Choosing the font of FormDesigner was enabled.
          Memorizing the position of InspectWindow was inabled.
          Accessor to childs was automatically added to parent control.
2004/03/03
          Improved so that it may not waste the resource of a font.
           FormDesiner 040303 released.
2004/03/20
          Updated with correction of swin and vruby.
2004/04/17
          Corrected the bug that the width and the height of ListBox were set
          to 0, when saved immediately after changing the style of ListBox.


fdvr/fdparser.rb
#
# DO NOT MODIFY!!!!
# This file is automatically generated by racc 1.4.4
# from racc grammer file "parse.y".
#

require 'racc/parser'


require 'fdvr/inner'


class FDParser < Racc::Parser

module_eval <<'..end parse.y modeval..iddbd93988db', 'parse.y', 66

include Inner
..end parse.y modeval..iddbd93988db

##### racc 1.4.4 generates ###

racc_reduce_table = [
 0, 0, :racc_error,
 1, 15, :_reduce_none,
 2, 15, :_reduce_none,
 1, 16, :_reduce_3,
 2, 16, :_reduce_4,
 1, 17, :_reduce_none,
 2, 17, :_reduce_none,
 1, 18, :_reduce_none,
 1, 18, :_reduce_none,
 1, 18, :_reduce_none,
 2, 22, :_reduce_10,
 3, 22, :_reduce_11,
 6, 22, :_reduce_12,
 4, 21, :_reduce_13,
 3, 21, :_reduce_14,
 5, 21, :_reduce_15,
 2, 21, :_reduce_16,
 3, 20, :_reduce_17,
 4, 20, :_reduce_18,
 1, 23, :_reduce_19,
 3, 23, :_reduce_20,
 1, 24, :_reduce_21,
 1, 24, :_reduce_none,
 1, 24, :_reduce_none,
 1, 24, :_reduce_none,
 2, 26, :_reduce_25,
 3, 25, :_reduce_26,
 2, 25, :_reduce_27,
 1, 27, :_reduce_28,
 3, 27, :_reduce_29,
 1, 28, :_reduce_30,
 1, 28, :_reduce_none,
 1, 19, :_reduce_none,
 1, 19, :_reduce_none,
 1, 19, :_reduce_none,
 1, 19, :_reduce_none ]

racc_reduce_n = 36

racc_shift_n = 55

racc_action_table = [
    18,    36,     5,     6,    39,    20,     6,    53,     5,     6,
    14,     1,     3,     4,     1,     3,     4,     1,     3,     4,
    29,   nil,    15,    16,   nil,    43,    33,    33,     1,     3,
     4,    33,    43,     1,     3,     4,    22,    23,    33,    48,
     1,     3,     4,    29,    15,    16,    31,    17,    29,    33,
   nil,     1,     3,     4,    33,    29,     1,     3,     4,   nil,
    29,    33,   nil,     1,     3,     4,    33,   nil,     1,     3,
     4,    22,    23,    51,    24,    52,    49,    40,    41,    41 ]

racc_action_check = [
     7,    18,     7,     7,    25,     9,     9,    49,     0,     0,
     2,     7,     7,     7,     9,     9,     9,     0,     0,     0,
    24,   nil,    29,    29,   nil,    51,    24,    29,    24,    24,
    24,    51,    33,    51,    51,    51,    39,    39,    33,    33,
    33,    33,    33,    16,     6,     6,    16,     6,    17,    16,
   nil,    16,    16,    16,    17,    41,    17,    17,    17,   nil,
    23,    41,   nil,    41,    41,    41,    23,   nil,    23,    23,
    23,    14,    14,    45,    14,    45,    37,    26,    37,    26 ]

racc_action_pointer = [
     6,   nil,     7,   nil,   nil,   nil,    40,     0,   nil,     3,
   nil,   nil,   nil,   nil,    67,   nil,    40,    45,     1,   nil,
   nil,   nil,   nil,    57,    17,     1,    71,   nil,   nil,    18,
   nil,   nil,   nil,    29,   nil,   nil,   nil,    70,   nil,    32,
   nil,    52,   nil,   nil,   nil,    65,   nil,   nil,   nil,     3,
   nil,    22,   nil,   nil,   nil ]

racc_action_default = [
   -36,   -33,   -36,   -34,   -35,    -3,   -32,   -36,    -1,   -36,
    -5,    -7,    -8,    -9,   -16,   -10,   -36,   -36,   -36,    -2,
    -4,    -6,   -11,   -36,   -36,   -36,   -36,   -19,   -22,   -32,
   -24,   -14,   -21,   -36,   -23,   -17,    55,   -36,   -18,   -16,
   -13,   -36,   -25,   -32,   -31,   -36,   -28,   -30,   -27,   -15,
   -20,   -36,   -26,   -12,   -29 ]

racc_goto_table = [
    11,    35,    42,    46,    13,     2,    44,    11,    38,    11,
    26,    13,     2,    13,     2,     8,    21,    37,    45,     7,
   nil,    54,    19,   nil,    44,    50,   nil,   nil,   nil,   nil,
   nil,   nil,   nil,    47,   nil,   nil,   nil,   nil,   nil,   nil,
   nil,   nil,   nil,   nil,   nil,   nil,   nil,   nil,   nil,   nil,
   nil,    47 ]

racc_goto_check = [
     5,    10,    11,    14,     7,     8,    11,     5,    10,     5,
     9,     7,     8,     7,     8,     2,     4,     9,    13,     1,
   nil,    14,     2,   nil,    11,    10,   nil,   nil,   nil,   nil,
   nil,   nil,   nil,     5,   nil,   nil,   nil,   nil,   nil,   nil,
   nil,   nil,   nil,   nil,   nil,   nil,   nil,   nil,   nil,   nil,
   nil,     5 ]

racc_goto_pointer = [
   nil,    19,    15,   nil,     7,     0,   nil,     4,     5,    -6,
   -16,   -27,   nil,   -15,   -30 ]

racc_goto_default = [
   nil,   nil,   nil,     9,    10,    32,    12,    34,    25,   nil,
    27,    28,    30,   nil,   nil ]

racc_token_table = {
 false => 0,
 Object.new => 1,
 "end" => 2,
 :IDNT => 3,
 "." => 4,
 "(" => 5,
 ")" => 6,
 "=" => 7,
 "," => 8,
 "[" => 9,
 "]" => 10,
 :NMBR => 11,
 :STRG => 12,
 :CNST => 13 }

racc_use_result_var = false

racc_nt_base = 14

Racc_arg = [
 racc_action_table,
 racc_action_check,
 racc_action_default,
 racc_action_pointer,
 racc_goto_table,
 racc_goto_check,
 racc_goto_default,
 racc_goto_pointer,
 racc_nt_base,
 racc_reduce_table,
 racc_token_table,
 racc_shift_n,
 racc_reduce_n,
 racc_use_result_var ]

Racc_token_to_s_table = [
'$end',
'error',
'"end"',
'IDNT',
'"."',
'"("',
'")"',
'"="',
'","',
'"["',
'"]"',
'NMBR',
'STRG',
'CNST',
'$start',
'accmtd',
'block',
'stlist',
'stmt',
'primary',
'assign',
'call',
'rcvr',
'args',
'arg',
'array',
'evalary',
'contents',
'field']

Racc_debug_parser = false

##### racc system variables end #####

 # reduce 0 omitted

 # reduce 1 omitted

 # reduce 2 omitted

module_eval <<'.,.,', 'parse.y', 14
  def _reduce_3( val, _values)
 do_end
  end
.,.,

module_eval <<'.,.,', 'parse.y', 15
  def _reduce_4( val, _values)
 do_end
  end
.,.,

 # reduce 5 omitted

 # reduce 6 omitted

 # reduce 7 omitted

 # reduce 8 omitted

 # reduce 9 omitted

module_eval <<'.,.,', 'parse.y', 24
  def _reduce_10( val, _values)
 do_send(val[0])
  end
.,.,

module_eval <<'.,.,', 'parse.y', 25
  def _reduce_11( val, _values)
 do_send2(val[0],val[1])
  end
.,.,

module_eval <<'.,.,', 'parse.y', 26
  def _reduce_12( val, _values)
	do_call(val[0],val[1],val[3])
  end
.,.,

module_eval <<'.,.,', 'parse.y', 28
  def _reduce_13( val, _values)
 do_call(nil,val[0],val[2])
  end
.,.,

module_eval <<'.,.,', 'parse.y', 29
  def _reduce_14( val, _values)
 do_call(nil,val[0],[])
  end
.,.,

module_eval <<'.,.,', 'parse.y', 30
  def _reduce_15( val, _values)
 do_call(val[0],val[1],val[3])
  end
.,.,

module_eval <<'.,.,', 'parse.y', 31
  def _reduce_16( val, _values)
 do_call(val[0],val[1],[])
  end
.,.,

module_eval <<'.,.,', 'parse.y', 33
  def _reduce_17( val, _values)
 do_assign(nil,val[0],val[2])
  end
.,.,

module_eval <<'.,.,', 'parse.y', 34
  def _reduce_18( val, _values)
 do_assign(val[0],val[1],val[3])
  end
.,.,

module_eval <<'.,.,', 'parse.y', 36
  def _reduce_19( val, _values)
 val
  end
.,.,

module_eval <<'.,.,', 'parse.y', 37
  def _reduce_20( val, _values)
 val[0].push val[2]
  end
.,.,

module_eval <<'.,.,', 'parse.y', 39
  def _reduce_21( val, _values)
 do_primary(val[0])
  end
.,.,

 # reduce 22 omitted

 # reduce 23 omitted

 # reduce 24 omitted

module_eval <<'.,.,', 'parse.y', 44
  def _reduce_25( val, _values)
 do_array(val[0],val[1])
  end
.,.,

module_eval <<'.,.,', 'parse.y', 46
  def _reduce_26( val, _values)
 val[1]
  end
.,.,

module_eval <<'.,.,', 'parse.y', 47
  def _reduce_27( val, _values)
 []
  end
.,.,

module_eval <<'.,.,', 'parse.y', 49
  def _reduce_28( val, _values)
 val
  end
.,.,

module_eval <<'.,.,', 'parse.y', 50
  def _reduce_29( val, _values)
 val[0].push val[2]
  end
.,.,

module_eval <<'.,.,', 'parse.y', 52
  def _reduce_30( val, _values)
 do_arr_prim(val[0])
  end
.,.,

 # reduce 31 omitted

 # reduce 32 omitted

 # reduce 33 omitted

 # reduce 34 omitted

 # reduce 35 omitted

 def _reduce_none( val, _values)
  val[0]
 end

end   # class FDParser
